lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -120,6 +120,22 @@ defmodule ExDoc.Formatter.HTML.Autolink do
     typespec_to_string(other, typespecs, aliases, lib_dirs)
   end
 
+  defp normalize_left({:::, _, [{name, _, args}, right]}, typespecs, aliases, lib_dirs) do
+    placeholder = {:::, [], [{:_, [], args}, right]}
+    "_" <> rest = typespec_to_string(placeholder, typespecs, aliases, lib_dirs)
+    Atom.to_string(name) <> rest
```

Comment:
Maybe what we could do is to first convert the args and then call Macro.to_string with the converted args as strings:

```iex
iex(2)> {name, meta, args} = {:foo, [], [1, 2, 3]}
{:foo, [], [1, 2, 3]}
iex(3)> new_args = Enum.map(args, &[self(), typespec_to_string(&1, typespecs, aliases, lib_dirs)])
iex(4)> Macro.to_string {name, meta, new_args}, fn [pid, string], _ when pid == self() -> string; _, string -> string end
"foo(1, 2, 3)"
```